File: VB/Q283154/Module1.vb
Imports Microsoft.VisualBasic
Imports Microsoft.Practices.Prism.MefExtensions.Modularity
Imports System.ComponentModel.Composition
Imports Microsoft.Practices.Prism.Modularity
Imports Microsoft.Practices.Prism.Regions
Namespace Q283154
	<ModuleExport(GetType(Module1))> _
	Public Class Module1
		Implements IModule
        <Import()> _
        Public RegionManager As IRegionManager

        Public Sub Initialize() Implements IModule.Initialize
            RegionManager.RegisterViewWithRegion("View1Region", GetType(View1))
        End Sub
    End Class
End Namespace

File: CS/Q283154/Module1.cs
using Microsoft.Practices.Prism.MefExtensions.Modularity;
using System.ComponentModel.Composition;
using Microsoft.Practices.Prism.Modularity;
using Microsoft.Practices.Prism.Regions;
namespace Q283154 {
    [ModuleExport(typeof(Module1))]
    public class Module1 : IModule {
        [Import]
        public IRegionManager RegionManager;
        public void Initialize() {
            RegionManager.RegisterViewWithRegion("View1Region", typeof(View1));
        }
    }
}